src/clndr.js

```diff
@@ -586,19 +604,7 @@
     };
     // did we click on a day or just an empty box?
     if(targetWasDay) {
-      var dateString;
-
-      // Our identifier is in the list of classNames. Find it!
-      var classNameIndex = currentTarget.className.indexOf('calendar-day-');
-      if(classNameIndex !== 0) {
-        // our unique identifier is always 23 characters long.
-        // If this feels a little wonky, that's probably because it is.
-        // Open to suggestions on how to improve this guy.
-        dateString = currentTarget.className.substring(classNameIndex + 13, classNameIndex + 23);
-        target.date = moment(dateString);
-      } else {
-        target.date = null;
-      }
+      var dateString = target.date = this.getTargetDate(currentTarget);
```

Comment:
I'm going to remove dateString here as I don't think it's needed here now that the function was moved out.